### PR TITLE
generate and upload coverage reports for dashboard

### PR DESCRIPTION
**Pull request by pcardune**
This PR changes the dashboard test helper to generate and upload coverage reports to codecov in circleci.

### CODE REVIEW

File: codecov.yml [hunk 1]
```diff
@@ -1,5 +1,6 @@
 comment: off
 coverage:
+  range: 50..80
```
Comment (islemaster) [inline codecov.yml:3]: Validated!
```
╰○ curl --data-binary @codecov.yml https://codecov.io/validate
Valid!

{
  "coverage": {
    "status": {
      "patch": false, 
      "project": false
    }, 
    "range": [
      50.0, 
      80.0
    ]
  }, 
  "comment": false
}
```

**Comment by pcardune** [comment]
@aoby at the moment, you will have to go to the codecov dashbaord and look up the report for the build in question. But it will look like this: https://codecov.io/gh/code-dot-org/code-dot-org/tree/f8fd4cb30d904a5aafc5d83aa67a8e93e0e6e045

**Comment by islemaster** [comment]
Where can I see this in action?

**Comment by pcardune** [comment]
You can see it in action here... https://codecov.io/gh/code-dot-org/code-dot-org

**Pull request merged**